prime-router/src/main/kotlin/metadata/LivdLookup.kt

```diff
@@ -0,0 +1,206 @@
+package gov.cdc.prime.router.metadata
+
+import gov.cdc.prime.router.Metadata
+
+// this will be used by the LivdMappers and the CustomFHIRFunction
+// /need to move most of LivdMappers in here
+// will need to move the tests to look at this as well
+// keep the library the way it was then use the context to use the library to pass in custom functions
+object LivdLookup {
+    fun find(
+        testPerformedCode: String?,
+        processingModeCode: String?,
+        deviceId: String?,
+        equipmentModelId: String?,
+        testKitNameId: String?,
+        equipmentModelName: String?,
+        tableColumn: String,
+        tableRef: LookupTable? = Metadata.getInstance().findLookupTable(name = "LIVD-SARS-CoV-2")
+    ): String? {
+        if (tableRef == null) {
+            return null
+        }
+
+        val filters = tableRef.FilterBuilder()
+        // get the test performed code for additional filtering of the test information in case we are
+        // dealing with tests that check for more than one type of disease, for example COVID + influenza
+        if (testPerformedCode != null) {
+            filters.equalsIgnoreCase(LivdTableColumns.TEST_PERFORMED_CODE.colName, testPerformedCode)
+        }
+
+        // If the data is NOT flagged as test data then ignore any test devices in the LIVD table
+        if (processingModeCode == null ||
+            processingModeCode.uppercase() != testProcessingModeCode
+        ) {
+            filters.notEqualsIgnoreCase(
+                LivdTableColumns.PROCESSING_MODE_CODE.colName,
+                testProcessingModeCode
+            )
+        }
+
+        // carry on as usual
+        val filtersCopy = filters.copy() // Filters are not reusable
+        return when {
+            !deviceId.isNullOrEmpty() -> lookupByDeviceId(
```

Comment:
This logic is not entirely right. I needs to be that it will always try the first lookup. If it is not successful then it will try the next. For example, say that I provided a deviceID, equipment model and test kit name.  All of it. Lookup by device ID. Did not find anything, then keep looking, if found then stop. Then look for model ID. Did not find anything, then keep looking, if found then stop. And so on.

Reply:
@PatriciaPerozoUSDS 
First clarification: the LIVD mapper takes in all pieces of data at the same time, so if all pieces are available then it will use all and do lookups until a match is found.
For the FHIR Path function, what we talked about was to take in the Observation and NOT the specific model name, or LOINC, or device name. Taking the observation will get us all the data available at once.

Reply:
We do want the same functionality as before

Reply:
Hmm ok maybe I'm not reading things properly but I couldn't find examples where we were actually sending more than one key at once. Happy to keep the same functionality as before though.